react/features/conference/components/native/Conference.js

```diff
@@ -192,10 +202,15 @@ class Conference extends AbstractConference<Props, State> {
      * @inheritdoc
      */
     componentDidUpdate(prevProps) {
-        const { _showLobby } = this.props;
+        const {
+            _localParticipantDisplayName,
+            _showLobby,
+            dispatch
+        } = this.props;
 
         if (!prevProps._showLobby && _showLobby) {
             navigate(screen.lobby.root);
+            _localParticipantDisplayName !== 'me' && dispatch(startKnocking());
```

Comment:
Use a checker that doesn't set a default to avoid checking for "me", since we will make that customizable at one point.

Also, use `shouldAutoKnock(state)` like web does.